pyteal/ast/abi/util.py

```diff
@@ -379,3 +389,87 @@ def algosdk_from_type_spec(t: TypeSpec) -> algosdk.abi.ABIType:
 
 def algosdk_from_annotation(t: type[T]) -> algosdk.abi.ABIType:
     return algosdk_from_type_spec(type_spec_from_annotation(t))
+
+
+def type_spec_from_algosdk(t: Union[algosdk.abi.ABIType, str]) -> TypeSpec:
+
+    from pyteal.ast.abi.reference_type import ReferenceTypeSpecs
+    from pyteal.ast.abi.transaction import TransactionTypeSpecs
+
+    from pyteal.ast.abi.array_dynamic import DynamicArrayTypeSpec
+    from pyteal.ast.abi.array_static import StaticArrayTypeSpec
+    from pyteal.ast.abi.tuple import TupleTypeSpec
+    from pyteal.ast.abi.bool import BoolTypeSpec
+    from pyteal.ast.abi.string import StringTypeSpec
+    from pyteal.ast.abi.address import AddressTypeSpec
+    from pyteal.ast.abi.uint import (
+        ByteTypeSpec,
+        Uint8TypeSpec,
+        Uint16TypeSpec,
+        Uint32TypeSpec,
+        Uint64TypeSpec,
+    )
+
+    match t:
+        case str():
```

Comment:
nit: would be better if you add a comment about sdk do not currently support reference types and transaction types, and they are handled only by `str`